public/samples/interop-samples/interop-samples.fs

```diff
@@ -0,0 +1,389 @@
+(*
+Interop Examples V 1.1 by Gary Bilkus March 2018
+
+This app is intended to demonstrate the various ways
+of interoperating with existing JavaScript.
+It goes progressively from a quick and dirty approach
+in which everything is untyped and you access the global
+namespace directly, through to use of modules and typed interfaces.
+
+The intention is to consolidate, in one reasonably small app,
+working examples of all the techniques available,
+from the perspective of someone who knows how to do things
+in JavaScript and wants to translate that into F#.
+
+The fable.io website does contain most of what's in here,
+but not necessarily in a way which is easy for a newbie to find.
+
+I am making no attempt to teach either JavaScript or F#.
+
+The index.html page which is the main page of this demo
+app contains a small script which defines various objects
+in JavaScript for us to use.
+
+http://fable.io/docs/compatibility.html
+and
+https://medium.com/@zaid.naom/f-interop-with-javascript-in-fable-the-complete-guide-ccc5b896a59f
+are (March 2018) the current main references, and worth reading.
+But there is some stuff in here which isn't in there,
+particularly the use of $ for function application
+and examples of Global, along with working examples
+of some things which are only briefly mentioned.
+*)
+namespace FableApp1
+(*
+Our first module demonstrates an approach to java interop
+in which we assume nothing at all about the underlying JavaScript
+objects, we just know how to use them from within existing JavaScript.
+
+We can do the same with Fable - the syntax is a bit different,
+but the principles are the same.
+*)
+module Utils =
+    open Fable.Core
+    open Fable.Import
+
+    // A nice way to show what's going on.....
+    [<Emit("JSON.stringify($0, null, 4) + ''")>]
+    let anyToString (_: obj) : string= jsNative
+
+    let code codeString value =
+        Browser.console.log("> " + codeString)
+        Browser.console.log(anyToString value + "\n")
+
+module VeryDirtyFable =
+    open Fable.Core
+    open Fable.Core.JsInterop
+    open Fable.Import
+    open Utils
+    // Suppose we have some JavaScript which creates
+    // some globals of various sorts
+    // We can import them by name as generic objects
+    // At this stage, we don't care what they really are
+    let [<Global("boringObj")>] boring1Obj : obj = jsNative
+    let [<Global("boringArr")>] boring1Arr : obj = jsNative
+    let [<Global("boringInt")>] boring1Int : obj = jsNative
+    let [<Global("boringString")>] boring1String : obj = jsNative
```

Comment:
@bilkusg Sorry, forgot to tell you that when importing a global function you can also type it as `Fable.Core.JsInterop.JsFunc`. Then you can invoke it dynamically later:

```fsharp
boring1String.Invoke(1, 2, "foo", "bar")
```